analyzers/src/SonarAnalyzer.CSharp/Rules/InterfacesShouldNotBeEmpty.cs

```diff
@@ -42,14 +42,33 @@ public sealed class InterfacesShouldNotBeEmpty : SonarDiagnosticAnalyzer
 
                     var interfaceSymbol = c.SemanticModel.GetDeclaredSymbol(interfaceDeclaration);
                     if (interfaceSymbol is { DeclaredAccessibility: Accessibility.Public }
-                        && !IsAggregatingOtherInterfaces(interfaceSymbol))
+                        && !IsAggregatingOtherInterfaces(interfaceSymbol)
+                        && !IsSpecializedGeneric(interfaceSymbol)
+                        && !HasEnhancingAttribute(interfaceSymbol))
                     {
                         c.ReportIssue(Diagnostic.Create(Rule, interfaceDeclaration.Identifier.GetLocation()));
                     }
                 },
                 SyntaxKind.InterfaceDeclaration);
 
         private static bool IsAggregatingOtherInterfaces(ITypeSymbol interfaceSymbol) =>
-            interfaceSymbol.AllInterfaces.Length > 1;
+            interfaceSymbol.Interfaces.Length > 1;
+
+        private static bool IsSpecializedGeneric(INamedTypeSymbol interfaceSymbol) =>
+            IsImplementingInterface(interfaceSymbol) && (IsBoundGeneric(interfaceSymbol) || IsConstraintGeneric(interfaceSymbol));
+
+        private static bool IsConstraintGeneric(INamedTypeSymbol interfaceSymbol) =>
+            interfaceSymbol.TypeParameters.Any(t => t.HasAnyConstraint());
```

Comment:
Based on our [coding style](https://github.com/SonarSource/sonar-dotnet/blob/master/docs/coding-style.md#naming-conventions):
```suggestion
            interfaceSymbol.TypeParameters.Any(x => x.HasAnyConstraint());
```